openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/OpenshiftHelmPushMojo.java

```diff
@@ -25,22 +26,15 @@
 public class OpenshiftHelmPushMojo extends HelmPushMojo {
 
   /**
-   * The generated kubernetes YAML file
-   */
-  @Parameter(property = "jkube.kubernetesManifest", defaultValue = "${basedir}/target/classes/META-INF/jkube/openshift.yml")
-  private File openShiftManifest;
-
-  /**
-   * The generated kubernetes YAML file
+   * One of:
+   * <ul>
+   * <li>A directory containing OpenShift Templates to use as Helm parameters.</li>
+   * <li>A file containing a Kubernetes List with OpenShift Template entries to be used as Helm parameters.</li>
+   * </ul>
    */
-  @Parameter(property = "jkube.kubernetesManifest", defaultValue = "${basedir}/target/classes/META-INF/jkube/openshift")
+  @Parameter(property = "jkube.openshiftTemplate", defaultValue = "${basedir}/target/classes/META-INF/jkube/openshift")
   private File openShiftTemplate;
 
-  @Override
-  protected File getKubernetesManifest() {
-    return openShiftManifest;
-  }
```

Comment:
I think it's there to override default location of manifests in the case of OpenShift Maven Plugin. 

Is the default value working for OpenShift?

Reply:
The manifest property only seems to be used in the `helm` goal, not in the `helm-push` goal. Because the `HelmMojo` class was both a base class and the implementation of the `helm` goal, this was not obvious previously.

Reply:
AFAIR helm push will likely need the overrides too to calculate the proper location of the package

Reply:
I do not see which override I am missing. But there are integration tests failing (which I can not yet run locally because I don't know how), so there might be something that needs overriding.

The `OpenshiftHelmPushMojo` extends from `HelmPushMojo` which extends from `AbstractHelmMojo`.

In `HelmPushMojo` there are only two functions (`init()` and `executeInternal()`) - I don't think I need to override those.
In `AbstractHelmMojo` there are the following functions:
- `init()`: I don't think I need to override this
- `getKubernetesTemplate()`: This is already overridden in `OpenshiftHelmPushMojo`
- `getDefaultHelmType()`: This is already overridden in `OpenshiftHelmPushMojo`
- `getHelm()`: I don't think I need to override this

Are you perhaps referring to `getKubernetesManifest()`? That method is only used in the `helm` goal. So in `HelmMojo` and in `OpenshiftHelmMojo` (where it is overridden).

But as I said: there are integration tests failing... Could you tell me how to run these locally?

Reply:
OK, I think everything in this conversation is outdated by now.

Things to be overridden in OpenShift version of mojos are:
```java
  @Override
  protected File getKubernetesManifest() {
    return openShiftManifest;
  }

  @Override
  protected File getKubernetesTemplate() {
    return openShiftTemplate;
  }

  @Override
  protected HelmConfig.HelmType getDefaultHelmType() {
    return HelmConfig.HelmType.OPENSHIFT;
  }

  @Override
  protected String getLogPrefix() {
    return OpenShift.DEFAULT_LOG_PREFIX;
  }
```

Reply:
Please see https://github.com/eclipse/jkube/pull/2714#discussion_r1511748520 for the continuation of this conversation. Let's mark this thread as resolved, and continue there.